controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -141,3 +144,82 @@ func (a *Adapter) getDeploymentTargetForEnvironment(environment *applicationapiv
 
 	return deploymentTarget, nil
 }
+
+// checkForBindingConditionsFailure will ensure that ephemeral environment(s) are cleaned up.
+func (a *Adapter) checkForBindingConditionsFailure() (reconciler.OperationResult, error) {
+	testEnvironment := a.environment
+
+	// get SnapshotEnvironmentBinding
+	binding, err := a.loader.FindExistingSnapshotEnvironmentBinding(a.client, a.context, a.application, testEnvironment)
```

Comment:
I don't think getting the binding here is necessary - this controller reacts to SnapshotEnvironmentBinding changes, so the `a.snapshotEnvironmentBinding`  will already contain the correct SnapshotEnvironmentBinding.